### PR TITLE
Enable readability-redundant-smartptr-get in clang-tidy

### PR DESCRIPTION
**Pull request by cyyever**

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -54,6 +54,7 @@ performance-*,
 readability-container-size-empty,
 readability-delete-null-pointer,
 readability-string-compare,
+readability-redundant-*,
```
Comment (Skylion007) [inline .clang-tidy:57]: please keep ordered if possible

Comment (cyyever) [reply .clang-tidy:57]: Fixed

**Comment by Skylion007** [comment]
While many of these rules are great, some are kind of unnecessary. After all, is redundant str init really so bad?

**Comment by cyyever** [comment]
> While many of these rules are great, some are kind of unnecessary. After all, is redundant str init really so bad?

In clang-tidy 18, the list is 
```
[readability-redundant-access-specifiers]
[readability-redundant-control-flow]
[readability-redundant-declaration]
[readability-redundant-function-ptr-dereference]
[readability-redundant-member-init]
[readability-redundant-preprocessor]
[readability-redundant-smartptr-get]
[readability-redundant-string-cstr]
[readability-redundant-string-init] 
```
IMO, readability-redundant-string-init, readability-redundant-access-specifiers are less useful.

**Comment by Skylion007** [comment]
Yeah, I would remove those less useful ones to start with probably.

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**